### PR TITLE
Speculative fix for #39

### PR DESCRIPTION
**Pull request by johnnyreilly**
See details in https://github.com/Realytics/fork-ts-checker-webpack-plugin/issues/39

### CODE REVIEW

**Comment by piotr-oles** [comment]
Great, thank you for your work! Just one thing is missing - could you add an integration test for it? :)

**Comment by johnnyreilly** [comment]
I have an idea for an integration test for this but it looks like tests don't run on Windows? Is that right?

**Comment by piotr-oles** [comment]
Oh, it's possible, I didn't run tests on Windows :/ Could you fix it? I think `NODE_ENV=test` is not used so you can remove it :) Also you can remove `node_modules/.bin/` prefix because npm/yarn will use this path automatically :)

**Comment by johnnyreilly** [comment]
No cigar :sad:  I've committed my speculative test - let's see what happens

**Comment by johnnyreilly** [comment]
The only way I can run this is by committing and letting Travis run it. The logic in my test is not quite correct - can you take a look please?  About to have to dash out the door

File: test/integration/index.spec.js [hunk 1]
```diff
@@ -185,4 +188,22 @@ describe('[INTEGRATION] index', function () {
       createCompiler({ tslint: true });
     }).to.not.throw.error;
   });
+
+  it('should not find syntactic errors when checkSyntacticErrors is false', function (callback) {
+    var compiler = createCompiler({ checkSyntacticErrors: false }, true);
+    
+    compiler.run(function(error, stats) {
+      expect(stats.compilation.errors.length).to.be.equal(2);
```
Comment (piotr-oles) [inline test/integration/index.spec.js]: I think it should be 1 :)

File: test/integration/index.spec.js [hunk 2]
```diff
@@ -185,4 +186,22 @@ describe('[INTEGRATION] index', function () {
...
+      callback();
+    });
+  });
+
+  it('should not find syntactic errors when checkSyntacticErrors is true', function (callback) {
```
Comment (piotr-oles) [inline test/integration/index.spec.js]: It should be `should find` not `should not find` :)

**Review by piotr-oles — CHANGES_REQUESTED** [review]

**Comment by johnnyreilly** [comment]
Yeah I'd appreciate some help with the test please 🌷  I've got to work now and not being able to run this on Windows is killing me.  The test I have in place *should* work but doesn't. And somehow I seem to have broken the `should find the same errors on multi-process mode` test but I can't see why.

The actual functionality of the change is fine AFAIK - I'm just failing to implement a test correctly.

**Comment by johnnyreilly** [comment]
Just implemented the changes requested.  I need help please. I'm leaving the house now and will hopefully be online again in about 3 hours....

File: test/integration/index.spec.js (hunk 1 above)
Comment (johnnyreilly) [reply test/integration/index.spec.js]: done

File: test/integration/index.spec.js (hunk 2 above)
Comment (johnnyreilly) [reply test/integration/index.spec.js]: quite right!

**Comment by piotr-oles** [comment]
Oki, I will fix tests :)

**Comment by johnnyreilly** [comment]
Thanks - I appreciate it 🌷

**Comment by johnnyreilly** [comment]
Taken from the issue, @voodoorider found this:

> @johnnyreilly It worked, however after linking your fork in `thread-loader` example I discovered that the syntactical errors are reported even without providing the `checkSyntacticErrors` flag.

This makes me think that the flag is always being set to `true` regardless.  This would explain why one of my tests is failing also (and perhaps the other one as well?)

I may have misunderstood how the options get passed through. I'm away from my laptop and can't run the code; I have tried tweaking just in browser.  If someone else gets the chance to take a look that'd be awesome.

**Comment by johnnyreilly** [comment]
OK it's just the pre-existing  `should find the same errors on multi-process mode` test that errors now.  Other problems fixed.

**Comment by johnnyreilly** [comment]
OK I think we're green.  Feel free to merge!

**Review by piotr-oles — APPROVED** [review]

**Comment by piotr-oles** [comment]
Thank you @johnnyreilly! Sorry I didn't help you with tests, I was very busy :/

**Comment by johnnyreilly** [comment]
No worries - got there in the end! Do you know when it will be released?

**Pull request merged**